scripts/metadata_comparison/metadata_comparison/comparer.py

```diff
@@ -4,101 +4,288 @@
+
+        for digester_key_name in digester_key_names:
+            digester_key = digester_key_by_json_key(digester_key_name)
+            top_header_row.append(f'{name_1} {digester_key.display_text}')
+            top_header_row.append(f'{name_2} {digester_key.display_text}')
+            top_header_row.append('% increase')
+
+            # Total times within a column, not necessarily total time of the job.
+            total_time_1, total_time_2 = [
+                sum_call_times(j, digester_key_name) for j in [json_1, json_2]]
+
+            percent_contrib_row.append(f'{(total_time_1 / total_total_time_1) * 100:.2f}%')
+            percent_contrib_row.append(f'{(total_time_2 / total_total_time_2) * 100:.2f}%')
+            percent_contrib_row.append('')
+
+            total_row.append(format_seconds(total_time_1))
+            total_row.append(format_seconds(total_time_2))
+            total_percent_increase = ((total_time_2 - total_time_1) * 100) / total_time_1
+            total_row.append(f'{total_percent_increase :.2f}%')
+
+            weighted_total_time_1, weighted_total_time_2 = [
+                sum_call_times_weighted(j, digester_key_name) for j in [json_1, json_2]]
+
+            machine_type_weighted_row.append(format_seconds(weighted_total_time_1))
+            machine_type_weighted_row.append(format_seconds(weighted_total_time_2))
+            weighted_percent = ((weighted_total_time_2 - weighted_total_time_1) * 100) / weighted_total_time_1
+            machine_type_weighted_row.append(f'{weighted_percent:.2f}%')
+
+        return [top_header_row, percent_contrib_row, total_row, machine_type_weighted_row, []]
+
+    rows = []
+
+    # Produce data for individual lifecycle phases of individual calls.
+    for call_key in call_keys_sorted_without_prefix:
+        row = []
+        call_1, call_2 = [j.get('calls').get(call_key.full) for j in [json_1, json_2]]
+        row.append(call_key.without_prefix)
+        row.append(call_1.get(MachineType))
+
+        for digester_key_name in digester_key_names:
+            time_1 = call_1.get(digester_key_name)
+            time_2 = call_2.get(digester_key_name)
+            row.append(format_seconds(time_1))
+            row.append(format_seconds(time_2))
+            if time_1:
+                row.append(f'{((time_2 - time_1)  * 100) / time_1:.2f}%')
+            else:
+                row.append('---')
+
+        rows.append(row)
+
+    return build_header_rows() + rows
+
+
+def error_checks(json_1: JsonObject, json_2: JsonObject, force: bool = False):
+    version_1, version_2 = [j.get('version') for j in [json_1, json_2]]
+
+    if version_1 != version_2:
+        msg = f"Inconsistent digest versions: First JSON digest is {version_1} but second is {version_2}"
+        raise ValueError(msg)
+
+    call_keys_1, call_keys_2 = [list(j.get('calls').keys()) for j in [json_1, json_2]]
+    call_keys_1.sort()
+    call_keys_2.sort()
+
+    if call_keys_1 != call_keys_2:
```

Comment:
[TOL] Could print out the set difference here to possibly expedite debugging this scenario (since we don't get to the `discrepancies` calculation)